app/views/user_mailer/reset_password_instructions.html.erb

```diff
@@ -33,7 +33,7 @@
               <td>
                 <center>
                   <%= link_to t('user_mailer.reset_password_instructions.link_text'),
-                              edit_user_password_url(reset_password_token: @token, locale: @locale),
+                              edit_user_password_url(reset_password_token: @token, locale: @locale, request_id: @request_id),
```

Comment:
can you provide some background on what linking the `request_id` does here? does it keep some session around that we need for attempts API?

If so, can we add some sort of acceptance spec that maybe clicks one of these emails and sees the event linked correctly? or something that helps explain why?

Reply:
Yes! I added this (some other emails also pass request_id in URL and they are logging events) to keep some SP sessions. 
Sure let me check I will try to add some tests.
![image](https://user-images.githubusercontent.com/109746710/216420415-526663f2-74ea-4585-b343-a2b858456f9b.png)

Reply:
currently, events get logged when the user clicks the reset password link from the email within the same browser session, but if a user closes the browser and then clicks the reset password link from the email our system does not log the event for that reason I added request_id in the links.